runtime/ms-rest/lib/filters/systemErrorRetryPolicyFilter.js

```diff
@@ -0,0 +1,167 @@
+    Math.floor(Math.random() * (this.retryInterval * 1.2 - this.retryInterval * 0.8));
+  incrementDelta *= boundedRandDelta;
+
+  retryData.retryInterval = Math.min(this.minRetryInterval + incrementDelta, this.maxRetryInterval);
+
+  return retryData;
+}
+
+
+/**
+ * Handles an operation with an exponential retry policy.
+ *
+ * @param {Object}   requestOptions The original request options.
+ * @param {function} next           The next filter to be handled.
+ * @return {undefined}
+ */
+function handle(requestOptions, next) {
+  var self = this;
+  var retryData = null;
+
+  var operation = function () {
+    // retry policies dont really do anything to the request options
+    // so move on to next
+    if (next) {
+      next(requestOptions, function (returnObject, finalCallback, nextPostCallback) {
+        // Previous operation ended so update the retry data
+        retryData = self.updateRetryData(retryData, returnObject.error);
+
+        if (!utils.objectIsNull(returnObject.error) && self.shouldRetry(retryData, returnObject.error) &&
+             (returnObject.error.code === 'ETIMEDOUT' || returnObject.error.code === 'ESOCKETTIMEDOUT' ||
+              returnObject.error.code === 'ECONNREFUSED' || returnObject.error.code === 'ECONNRESET' ||
+              returnObject.error.code === 'ENOTFOUND')) {
+          // If previous operation ended with an error and the policy allows a retry, do that
+          setTimeout(function () {
+            operation();
+          }, retryData.retryInterval);
+        } else {
+          if (!utils.objectIsNull(returnObject.error)) {
+            // If the operation failed in the end, return all errors instead of just the last one
+            returnObject.error = retryData.error;
+          }
+          if (nextPostCallback) {
+            nextPostCallback(returnObject);
+          } else if (finalCallback) {
+            finalCallback(returnObject);
+          }
+        }
+      });
+    }
+  };
+
+  operation();
+}
+
+/**
+ * Creates a new 'SystemErrorRetryPolicyFilter' instance.
+ *
+ * @constructor
+ * @param {number} retryCount        The client retry count.
+ * @param {number} retryInterval     The client retry interval, in milliseconds.
+ * @param {number} minRetryInterval  The minimum retry interval, in milliseconds.
+ * @param {number} maxRetryInterval  The maximum retry interval, in milliseconds.
+ */
+function SystemErrorRetryPolicyFilter(retryCount, retryInterval, minRetryInterval, maxRetryInterval) {
+  function newFilter(options, next, callback) {
+    var retryData = null;
+
+    function retryCallback(err, response, body) {
+      retryData = newFilter.updateRetryData(retryData, err);
+      if (!utils.objectIsNull(err) && newFilter.shouldRetry(retryData, err) && 
+          (err.code === 'ETIMEDOUT' || err.code === 'ESOCKETTIMEDOUT' || err.code === 'ECONNREFUSED' || 
+           err.code === 'ECONNRESET' || err.code === 'ENOTFOUND')) {
```

Comment:
@huangpf - This is the new filter that will retry for some of the selected system Errors. This should solve the issue reported here in xplat cli https://github.com/Azure/azure-xplat-cli/issues/3073